app/samples/phase_detection/src/main.cpp

```diff
@@ -0,0 +1,241 @@
+// F-Core Includes
+#include "f_core/util/debouncer.hpp"
+#include "f_core/util/linear_fit.hpp"
+
+#include <f_core/device/sensor/c_accelerometer.h>
+#include <f_core/device/sensor/c_barometer.h>
+#include <f_core/device/sensor/c_gyroscope.h>
+#include <f_core/device/sensor/c_magnetometer.h>
+#include <math.h>
+#include <zephyr/kernel.h>
+
+// Zephyr Includes
+#include "flight.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(main, CONFIG_APP_PHASE_DETECT_LOG_LEVEL);
+
+#include "f_core/util/debouncer.hpp"
+
+#include <array>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+
+K_TIMER_DEFINE(imu_timer, NULL, NULL);
+K_TIMER_DEFINE(barom_timer, NULL, NULL);
+
+Controller controller{sourceNames, eventNames, timer_events, deciders};
```

Comment:
Actually create the controller device. Doesn't have  to be static, could be created in main and just pass references around but im lazy and didnt want to dereference a bunch of void pointers in threads.